pkg/local_workflows/data_transformation_workflow.go

```diff
@@ -108,63 +108,353 @@ func dataTransformationEntryPoint(invocationCtx workflow.InvocationContext, inpu
+						LineNumber int    `json:"lineNumber"`
+						Linechange string `json:"linechange"`
+					} `json:"lines"`
+				} `json:"exampleCommitFixes"`
+				Precision       string   `json:"precision"`
+				RepoDatasetSize int      `json:"repoDatasetSize"`
+				Tags            []string `json:"tags"`
+			}{
+				Categories:                rule.Properties.Categories,
+				Cwe:                       rule.Properties.Cwe,
+				ExampleCommitDescriptions: rule.Properties.ExampleCommitDescriptions,
+				ExampleCommitFixes: func() []struct {
+					CommitUrl string `json:"commitUrl"`
+					Lines     []struct {
+						Line       string `json:"line"`
+						LineNumber int    `json:"lineNumber"`
+						Linechange string `json:"linechange"`
+					} `json:"lines"`
+				} {
+					var fixes []struct {
+						CommitUrl string `json:"commitUrl"`
+						Lines     []struct {
+							Line       string `json:"line"`
+							LineNumber int    `json:"lineNumber"`
+							Linechange string `json:"linechange"`
+						} `json:"lines"`
+					}
+					for _, fix := range rule.Properties.ExampleCommitFixes {
+						fixes = append(fixes, struct {
+							CommitUrl string `json:"commitUrl"`
+							Lines     []struct {
+								Line       string `json:"line"`
+								LineNumber int    `json:"lineNumber"`
+								Linechange string `json:"linechange"`
+							} `json:"lines"`
+						}{
+							CommitUrl: fix.CommitURL,
+							Lines: func() []struct {
+								Line       string `json:"line"`
+								LineNumber int    `json:"lineNumber"`
+								Linechange string `json:"linechange"`
+							} {
+								var lines []struct {
+									Line       string `json:"line"`
+									LineNumber int    `json:"lineNumber"`
+									Linechange string `json:"linechange"`
+								}
+								for _, line := range fix.Lines {
+									lines = append(lines, struct {
+										Line       string `json:"line"`
+										LineNumber int    `json:"lineNumber"`
+										Linechange string `json:"linechange"`
+									}{
+										Line:       line.Line,
+										LineNumber: line.LineNumber,
+										Linechange: line.LineChange,
+									})
+								}
+								return lines
+							}(),
+						})
+					}
+					return fixes
+				}(),
+				Precision:       rule.Properties.Precision,
+				RepoDatasetSize: rule.Properties.RepoDatasetSize,
+				Tags:            rule.Properties.Tags,
+			},
+		})
 	}
+	return rules
+}
 
-	localFinding.Summary.Path = testSummary.Path
-	localFinding.Summary.Artifacts = testSummary.Artifacts
-	localFinding.Summary.Type = testSummary.Type
-	localFinding.Summary.Counts.CountKeyOrderAsc.Severity = testSummary.SeverityOrderAsc
-	localFinding.Summary.Counts.Count = 0
-	localFinding.Summary.Counts.CountAdjusted = 0
-	localFinding.Summary.Counts.CountSuppressed = 0
+func transformTestSummary(testSummary *json_schemas.TestSummary, sarifDoc *sarif.SarifDocument) *local_models.TypesFindingsSummary {
+	var summary local_models.TypesFindingsSummary
```

Comment:
Suggestion: look into using the existing [NewFindingsCounts()](https://github.com/snyk/go-application-framework/blob/main/internal/utils/findings/findings.go#L33)